packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx

```diff
@@ -64,6 +65,21 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
     sourceIndex: number;
     targetIndex: number;
   };
+
+  public static getDerivedStateFromProps(newProps: IDetailsHeaderBaseProps, prevState: IDetailsHeaderState): IDetailsHeaderState {
+    const columnReorderProps: IColumnReorderHeaderProps | undefined =

```

Comment:
`columnReorderProps` probably shouldn't be in `state` either, since it's never updated with `setState`. A helper function `getColumnReorderProps` (ideally memoized) would be more appropriate.

The tricky part with memoizing is that our current memoize functions won't work as expected with objects--it will only return the cached result for the same parent object identity, rather than checking the members. So ideally you should pass the relevant props as individual parameters, rather than passing the props object itself (though even passing the props object would be better than having no memoization).

```ts
const getColumnReorderProps = memoizeFunction((
  columnReorderProps: IColumnReorderHeaderProps | undefined, 
  columnReorderOptions: IColumnReorderOptions | undefined
): IColumnReorderHeaderProps | undefined => {
  return columnReorderProps || (columnReorderOptions && {
    ...columnReorderOptions,
    onColumnDragEnd: undefined
  });
});
```
or
```ts
const getColumnReorderProps = memoizeFunction((
  props: IDetailsHeaderBaseProps
): IColumnReorderHeaderProps | undefined => {
  const { columnReorderProps, columnReorderOptions } = props;
  return columnReorderProps || (columnReorderOptions && {
    ...columnReorderOptions,
    onColumnDragEnd: undefined
  });
});
```